codegen-test/model/rest-json-extras.smithy

```diff
@@ -322,9 +335,118 @@ structure EmptyStructWithContentOnWireOpOutput {
         protocol: "aws.protocols#restJson1",
         code: 200,
         body: "{\"empty\": {\"value\":\"not actually empty\"}}",
-        params: { empty: {} }
+        params: { empty: {} },
+        appliesTo: "client",
     }
 ])
 operation EmptyStructWithContentOnWireOp {
     output: EmptyStructWithContentOnWireOpOutput,
 }
+
+apply FixedMalformedAcceptWithGenericString @httpMalformedRequestTests([
+    {
+        id: "RestJsonWithPayloadExpectsImpliedAcceptFixed",
+        documentation: """
+        When there is a payload without a mediaType trait, the accept must match the
+        implied content type of the shape.""",
+        protocol: restJson1,
+        request: {
+            method: "POST",
+            uri: "/FixedMalformedAcceptWithGenericString",
+            headers: {
+                // this should be text/plain
+                "accept": "application/json"
+            }
+        },
+        response: {
+            code: 406,
+            headers: {
+                "x-amzn-errortype": "NotAcceptableException"
+            }
+        },
+        tags: [ "accept" ],
+        appliesTo: "server",
+    }
+])
+
+@suppress(["UnstableTrait"])
+@http(method: "POST", uri: "/FixedMalformedAcceptWithGenericString")
+operation FixedMalformedAcceptWithGenericString {
+    input: FixedMalformedAcceptWithGenericStringInput
+}
+
+structure FixedMalformedAcceptWithGenericStringInput {
+    @httpPayload
+    payload: String
+}
+
+@idempotent
+@http(method: "PUT", uri: "/service/{id}")
+@documentation("Service register operation")
+@httpRequestTests([
+    {
+        id: "FixedRegisterServiceRequestTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/service/1",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        params: { id: "1", name: "TestService" },
+        body: "{\"name\":\"TestService\"}",
+        method: "PUT",
+        appliesTo: "server",
+    }
+])
+@httpResponseTests([
+    {
+        id: "FixedRegisterServiceResponseTest",
+        protocol: "aws.protocols#restJson1",
+        params: { id: "1", name: "TestService" },
+        headers: {
+            "Content-Type": "TestService",
+        },
+        body: "{\"id\":\"1\"}",
+        code: 200,
+        appliesTo: "server",
+    }
+])
+operation FixedRegisterService {
```

Comment:
What are these tests testing that is not already being tested in the `rest_json` suite?

Reply:
For #1164, but will address that in another PR